### PR TITLE
query users color from ulib table in connect hook

### PR DESCRIPTION
**Pull request by plally**

### CODE REVIEW

File: lua/autorun/server/sv_apple.lua [hunk 1]
```diff
@@ -15,6 +30,7 @@ local function onPlayerConnect( data )
 
     net.Start( "cfc_playerconnect_ajl" )
         net.WriteString( plyName )
+        net.WriteColor( getOfflineColor(data.networkid) or prefixColor )
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_apple.lua]:
```suggestion
        net.WriteColor( getOfflineColor( data.networkid ) or prefixColor )
```

File: lua/autorun/server/sv_apple.lua [hunk 2]
```diff
@@ -7,6 +7,21 @@ util.AddNetworkString( "cfc_playerdisconnect_ajl" )
...
 local serverStarTime = CurTime() -- Likely near 1 always, but worth to set anyway
 
+local function getOfflineColor(steamID32)
+    local ply = ULib.ucl.users[steamID32]
+    if not ply then return end
+    
+    local group = ULib.ucl.groups[ply.group or ""]
+    if not group then return end
+
+    if not (group.team.color_red and group.team.color_green and group.team.color_red) then return end
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_apple.lua]:
```suggestion
    if not group.team.color_red then return end
    if not group.team.color_green then return end
    if not group.team.color_red then return end
```

File: lua/autorun/server/sv_apple.lua [hunk 3]
```diff
@@ -7,6 +7,21 @@ util.AddNetworkString( "cfc_playerdisconnect_ajl" )
...
+    
+    local color = Color(tonumber(group.team.color_red), tonumber(group.team.color_green), tonumber(group.team.color_blue))
```
Comment (sarahsturgeon) [inline lua/autorun/server/sv_apple.lua]:
```suggestion
    local color = Color(
        tonumber( group.team.color_red ),
        tonumber( group.team.color_green ),
        tonumber( group.team.color_blue )
    )
```

**Review by sarahsturgeon — APPROVED** [review]
LGTM

Just a couple of spacing/formatting suggestions. One practical question, though.

File: lua/autorun/client/cl_apple.lua [hunk 4]
```diff
@@ -14,8 +14,8 @@ end
 
 net.Receive( "cfc_playerconnect_ajl", function()
     local name = net.ReadString()
-
-    chat.AddText( prefixColor, "• ", textColor, name .. " has connected to the server." )
+    local color = net.ReadColor()
+    chat.AddText( color, "• ", textColor, name .. " has connected to the server." )
```
Comment (sarahsturgeon) [inline lua/autorun/client/cl_apple.lua]: Wouldn't we just want their name to be rank-colored?
```suggestion
    chat.AddText( prefixColor, "• ", color, name, textColor, " has connected to the server." )
```

**Review by sarahsturgeon — DISMISSED** [review]

File: lua/autorun/server/sv_apple.lua (hunk 2 above)
Comment (plally) [reply lua/autorun/server/sv_apple.lua]: egh, i think this is fine now that i localized team.

File: lua/autorun/client/cl_apple.lua (hunk 4 above)
Comment (plally) [reply lua/autorun/client/cl_apple.lua]: good catch

File: lua/autorun/server/sv_apple.lua (hunk 2 above)
Comment (sarahsturgeon) [reply lua/autorun/server/sv_apple.lua]: just a long line is all

**Review by sarahsturgeon — APPROVED** [review]

**Pull request merged**